GitUI/UserControls/RevisionGrid/Columns/RevisionGraphColumnProvider.cs

```diff
@@ -199,52 +198,51 @@ bool DrawVisibleGraph()
 
                         _graphCache.GraphBitmapGraphics.RenderingOrigin = new Point(x, y);
 
-                        var success = DrawItem(_graphCache.GraphBitmapGraphics, index);
+                        DrawItem(_graphCache.GraphBitmapGraphics, index);
 
                         _graphCache.GraphBitmapGraphics.Clip = oldClip;
-
-                        if (!success)
-                        {
-                            _graphCache.Reset();
-                            return false;
-                        }
                     }
-
-                    return true;
                 }
 
-                bool DrawItem(Graphics g, int index)
+                void DrawItem(Graphics g, int index)
                 {
-                    // Clip to the area we're drawing in, but draw 1 pixel past so
-                    // that the top/bottom of the line segment's anti-aliasing isn't
-                    // visible in the final rendering.
-                    int top = g.RenderingOrigin.Y; //// + rowHeight;
-                    var laneRect = new Rectangle(0, top, width, rowHeight);
+                    SmoothingMode oldSmoothingMode = g.SmoothingMode;
                     Region oldClip = g.Clip;
-                    var newClip = new Region(laneRect);
+
+                    int top = g.RenderingOrigin.Y;
+                    Rectangle laneRect = new(0, top, width, rowHeight);
+                    Region newClip = new(laneRect);
                     newClip.Intersect(oldClip);
                     g.Clip = newClip;
                     g.Clear(Color.Transparent);
```

Comment:
Why do we need to use `Clear` vs. `FillRegion`? The latter would avoid needing to set/restore the `Clip`.

Reply:
`Clip` is needed anyway in order to create the impression of drawn through graph lines by not letting end the lines at the row borders.